ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -816,13 +816,19 @@ def _get_connections_core(
             instrument_type_id, self._reserved_instrument_type_ids
         )
 
+        requested_pin_or_relay_names = _to_ordered_set(
```

Comment:
With this update, the `get_connections` API now ignores duplicate connections. For instance, when requesting connections for ['Pin1', 'Pin1'], the current implementation returns only one connection for Pin1. In the previous version, the get-connection(s) API would return two connections with the same Pin1 data. 

Please confirm whether there is a valid reason for returning duplicate data, or if the updated change to ignore duplicates is acceptable.

Cc: @bkeryan, @dixonjoel

Reply:
I don't know of any valid reason.

Reply:
We're removing the duplicates from the resolved requested pin list. Therefore, the duplicate pins will be removed before framing the connection response.

Reply:
> We're removing the duplicates from the resolved requested pin list.

This is because of pin groups, right? Because they may overlap?

Reply:
> This is because of pin groups, right? Because they may overlap?

yes.


> I'm not asking about what you're doing. I'm asking about why you're doing it.

I apologize. As you mentioned, to avoid overlaps, we are removing duplicates.

Reply:
ok, thanks